roles/validate_inventory/tasks/ai.yml

```diff
@@ -28,3 +28,36 @@
     fail_msg: "{{ item }} is not within the machine network!"
   when: vip_dhcp_allocation == false
   loop: "{{ groups['masters'] + (groups['workers'] | default([])) }}"  # This should not include day2_workers as they can be RWNs
+
+- name: Validate extra VIPs for dualstack
+  block:
+    - name: Assert that Openshift version is supported for dualstack VIPs
+      assert:
+        that:
+          - openshift_full_version is version('4.12', '>=')
+        fail_msg: "openshift_full_version  must be >= 4.12. to support dualstack VIPs"
+
+    - name: Assert extra api VIP is within the extra machine networks
+      assert:
+        that:
+          - hostvars['assisted_installer']['extra_api_vip'] | ansible.utils.ipaddr(item.cidr) | ansible.utils.ipaddr('bool')
+        fail_msg: "{{ extra_api_vip }} is not within the extra machine networks!"
+      when:
+        - vip_dhcp_allocation == false
+        - extra_machine_networks is defined
+      loop: "{{ hostvars['assisted_installer']['extra_machine_networks'] }}"
+
+    - name: Assert extra ingress VIP is within the extra machine networks
+      assert:
+        that:
+          - hostvars['assisted_installer']['extra_ingress_vip'] | ansible.utils.ipaddr(item.cidr) | ansible.utils.ipaddr('bool')
+        fail_msg: "{{ extra_ingress_vip }} is not within the extra machine networks!"
+      when:
+        - vip_dhcp_allocation == false
+        - extra_machine_networks is defined
+      loop: "{{ hostvars['assisted_installer']['extra_machine_networks'] }}"
+  when:
```

Comment:
Can you put `when` above `block` it makes the block easier to read.

Reply:
Done, added an extra commit